File: usaco/2020Jan/Race.py
import math

with open("race.in") as f:
    raw = f.readlines()
    k, n = tuple(map(int, raw[0].split()))
    xs = list(map(int, raw[1:]))

def findTime(x):
    remDist = k
    speed = 1
    time = 0
    while True:
        remDist -= speed
        time += 1
        if remDist <= 0:
            return time
        if speed >= x:
            remDist -= speed
            time += 1
            if remDist <= 0:
                return time
        speed += 1

with open("race.out", "w+") as f:
    for x in xs:
        f.write(str(findTime(x)) + '\n')

File: usaco/2020Jan/race.java
import java.util.Scanner;
import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.io.Writer;
import java.nio.file.Paths;


public class race {
    public static int findTime(int x, int k) {
        int speed = 1;
        int time = 0;
        while (true) {
            k -= speed;
            time++;
            if (k <= 0) {
                return time;
            }
            if (speed >= x) {
                k -= speed;
                time++;
                if (k <= 0) {
                    return time;
                }
            }
            speed++;
        }
    }
    public static void main(String[] args) throws IOException {
        Scanner in = new Scanner(new File(args[0]));
        int k = Integer.parseInt(in.next());
        int n = Integer.parseInt(in.next());
        int[] xs = new int[n];
        for (int i = 0; i < n; i++) {
            xs[i] = Integer.parseInt(in.next());
        }
        in.close();

        Writer out = Files.newBufferedWriter(Paths.get(args[1]));
        for (int x : xs) {
            out.write(findTime(x, k));
            out.write("\n");
        }
        out.close();
    }
}